### PR TITLE
Paircorr update

### PR DESCRIPTION
**Pull request by cpgoodri**
This PR makes two small additions to the pair correlation functionality

1. Slightly improves the documentation for `quantity.pair_correlation` and `quantity.pair_correlation_neighbor_list`
2. Adds a function `quantity.average_pair_correlation_results` that takes the results of the two pair correlation functions and averages over particles of the same species. When not using species, this is simply `jnp.mean(results, axis=0)`, but for species you typically want averages of the form $g_{AB}(r)$ for all pairs of species $A$ and $B$. This function pulls out all these averages. 

Thoughts/suggestions?

### CODE REVIEW

File: jax_md/quantity.py [hunk 1]
```diff
@@ -285,6 +285,39 @@ def is_integer(x: Array) -> bool:
   return x.dtype == jnp.int32 or x.dtype == jnp.int64
 
 
+def average_pair_correlation_results(gofr, species=None):
```
Comment (sschoenholz) [inline jax_md/quantity.py:288]: This seems reasonable, but we could also add a keyword argument to `pair_correction` that says "compute_average` or something? Is there a reason you went with this approach?

To be honest, it could be nice to have both. Keep this function and then add a convenience keyword argument that averages after computing.

Comment (cpgoodri) [reply jax_md/quantity.py:288]: I was actually thinking along the same lines. Just added it. Let me know if you have any other comments.
